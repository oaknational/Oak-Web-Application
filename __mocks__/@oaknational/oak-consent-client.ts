const consentClientMock = jest.requireActual("@oaknational/oak-consent-client");

/**
 * Provides a mock implementation of OakConsentClient so that tests
 * do not try to hit the network
 */
class MockOakConsentClient {
  getState() {
    return {
      policyConsents: [],
      requireInteraction: false,
    };
  }
  logConsents() {
    return Promise.resolve();
  }
  getConsent() {
    return "pending";
  }
  onStateChange() {
    return () => {};
  }
}

module.exports = {
  ...consentClientMock,
  OakConsentClient: MockOakConsentClient,
};
