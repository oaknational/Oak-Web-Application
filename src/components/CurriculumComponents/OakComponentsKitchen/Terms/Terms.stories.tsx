import { OakThemeProvider, oakDefaultTheme } from "@oaknational/oak-components";
import { Meta, StoryObj } from "@storybook/react";

import Component from ".";

const meta: Meta<typeof Component> = {
  component: Component,
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof Component>;

export const Terms: Story = {
  args: {},
  render: () => {
    return (
      <OakThemeProvider theme={oakDefaultTheme}>
        <Component />
      </OakThemeProvider>
    );
  },
};
