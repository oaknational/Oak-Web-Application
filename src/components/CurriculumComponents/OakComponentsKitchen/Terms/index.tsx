import { OakLink, OakP } from "@oaknational/oak-components";
import { ComponentProps } from "react";

import { resolveOakHref } from "@/common-lib/urls";

const Link = (props: ComponentProps<typeof OakLink>) => {
  const isExternal = props.rel === "external";
  const additionalProps = isExternal
    ? ({
        target: "_blank",
      } as const)
    : {};
  return (
    <OakLink {...props} {...additionalProps} style={{ display: "inline" }}>
      {props.children}
    </OakLink>
  );
};

export default function Terms() {
  return (
    <OakP $font={["body-3"]}>
      This content is Â© Oak National Academy (2023), licensed on{" "}
      <Link
        target="_blank"
        rel="external"
        href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
      >
        Open Government Licence version 3.0
      </Link>{" "}
      except where otherwise stated. See{" "}
      <Link
        rel="external"
        href={resolveOakHref({
          page: "legal",
          legalSlug: "terms-and-conditions",
        })}
      >
        Oak's terms and conditions
      </Link>
      .
    </OakP>
  );
}
