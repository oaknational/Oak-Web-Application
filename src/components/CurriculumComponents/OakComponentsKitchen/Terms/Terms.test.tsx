import { oakDefaultTheme, OakThemeProvider } from "@oaknational/oak-components";

import Terms from ".";

import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";

test("<Terms/>", () => {
  const { baseElement } = renderWithTheme(
    <OakThemeProvider theme={oakDefaultTheme}>
      <Terms />
    </OakThemeProvider>,
  );
  expect(baseElement).toMatchSnapshot();
});
