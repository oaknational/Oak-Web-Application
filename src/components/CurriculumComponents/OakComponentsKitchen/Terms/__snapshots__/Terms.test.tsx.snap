// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Terms/> 1`] = `
<body>
  <div>
    <p
      class="sc-dLMFU MdFoC"
    >
      This content is Â© Oak National Academy (2023), licensed on
       
      <a
        class="sc-ddjGPC hJhpMe"
        href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
        rel="external"
        style="display: inline;"
        target="_blank"
      >
        <span
          class="sc-dAlyuH gAWMRU"
        >
          Open Government Licence version 3.0
        </span>
      </a>
       
      except where otherwise stated. See
       
      <a
        class="sc-ddjGPC hJhpMe"
        href="/legal/terms-and-conditions"
        rel="external"
        style="display: inline;"
        target="_blank"
      >
        <span
          class="sc-dAlyuH gAWMRU"
        >
          Oak's terms and conditions
        </span>
      </a>
      .
    </p>
  </div>
</body>
`;
